### PR TITLE
Inject required args only into plugins config

### PR DESCRIPTION
**Pull request by collijk**
### Changes
- Add a `_safe_split` function to reduce some repeated code
- Inject only the required arguments into the plugin config so we don't need to wait on any complex state initialization
- initialize the plugin config when we're initializing the rest of the `Config` arguments so we're not doing special case ordering.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run `black .` and `isort .` against my code to ensure it passes our linter.

### CODE REVIEW

File: autogpt/config/config.py [hunk 1]
```diff
@@ -83,11 +83,6 @@ class Config(SystemSettings):
     plugins: list[str]
     authorise_key: str
 
-    # Executed immediately after init by Pydantic
-    def model_post_init(self, **kwargs) -> None:
```
Comment (collijk) [inline autogpt/config/config.py:87]: No longer needed as we initialize the `PluginsConfig` with the rest of the `Config` arguments

File: autogpt/config/config.py [hunk 2]
```diff
@@ -206,21 +201,16 @@ def build_config_from_env(cls) -> Config:
             "chat_messages_enabled": os.getenv("CHAT_MESSAGES_ENABLED") == "True",
         }
 
-        # Converting to a list from comma-separated string
```
Comment (collijk) [inline autogpt/config/config.py:209]: This just using `_safe_split` to reduce repeated logic.

File: autogpt/config/config.py [hunk 3]
```diff
@@ -230,13 +220,13 @@ def build_config_from_env(cls) -> Config:
...
-        plugins_allowlist = os.getenv("ALLOWLISTED_PLUGINS")
-        if plugins_allowlist:
-            config_dict["plugins_allowlist"] = plugins_allowlist.split(",")
-
-        plugins_denylist = os.getenv("DENYLISTED_PLUGINS")
-        if plugins_denylist:
-            config_dict["plugins_denylist"] = plugins_denylist.split(",")
+        config_dict["plugins_allowlist"] = _safe_split(os.getenv("ALLOWLISTED_PLUGINS"))
+        config_dict["plugins_denylist"] = _safe_split(os.getenv("DENYLISTED_PLUGINS"))
+        config_dict["plugins_config"] = PluginsConfig.load_config(
```
Comment (collijk) [inline autogpt/config/config.py:225]: This is the only real change in this function

File: autogpt/core/configuration/schema.py [hunk 4]
```diff
@@ -49,14 +49,7 @@ def build_agent_configuration(cls, configuration: dict) -> S:
         defaults = cls.default_settings.dict()
         final_configuration = deep_update(defaults, configuration)
 
-        config_obj = cls.default_settings.__class__.parse_obj(final_configuration)
```
Comment (collijk) [inline autogpt/core/configuration/schema.py:52]: No longer needed as `Config` has no `model_post_init`

File: autogpt/plugins/plugins_config.py [hunk 5]
```diff
@@ -30,11 +26,20 @@ def is_enabled(self, name) -> bool:
         return plugin_config is not None and plugin_config.enabled
 
     @classmethod
-    def load_config(cls, global_config: Config) -> "PluginsConfig":
+    def load_config(
```
Comment (collijk) [inline autogpt/plugins/plugins_config.py:29]: There are no logic changes in this file, just unpacked the arguments.

**Review by Pwuts — APPROVED** [review]

File: autogpt/config/config.py (hunk 3 above)
Comment (Pwuts) [reply autogpt/config/config.py:225]: Looks neat :)

**Pull request merged**